src/public/modules/forms/admin/components/response-components/response-attachment.client.component.js

```diff
@@ -24,12 +24,13 @@ function responseAttachmentComponentController($q, $timeout) {
       ),
     )
       .then((bytesArray) => {
-        if (bytesArray) {
-          // Construct a downloadable link and click on it to download the file
-          let blob = new Blob([bytesArray])
-          // field.answer is the filename
-          triggerFileDownload(blob, vm.field.answer)
+        if (!bytesArray) {
+          vm.hasDownloadError = true
```

Comment:
does this work even without the `$timeout` (as in the `catch` block)? it seems that the digest cycle should be triggered by $q.when but would be good to double check

Reply:
good catch - tested it and it works!